File: backend/core/src/main/java/site/timecapsulearchive/core/domain/capsule/group_capsule/data/dto/GroupCapsuleMemberDto.java
package site.timecapsulearchive.core.domain.capsule.group_capsule.data.dto;

import site.timecapsulearchive.core.domain.capsule.group_capsule.data.response.GroupCapsuleMemberResponse;

public record GroupCapsuleMemberDto(
    Long id,
    String nickname,
    String profileUrl,
    Boolean isGroupOwner,
    Boolean isOpened
) {

    public GroupCapsuleMemberResponse toResponse() {
        return new GroupCapsuleMemberResponse(id, nickname, profileUrl, isGroupOwner, isOpened);
    }
}

File: frontend/ARchive/app/src/main/java/com/droidblossom/archive/data/dto/group_capsule/GroupCapsuleMemberDto.kt
package com.droidblossom.archive.data.dto.group_capsule

import com.droidblossom.archive.domain.model.group_capsule.GroupCapsuleMember

data class GroupCapsuleMemberDto(
    val id: Long,
    val nickname : String,
    val profileUrl : String,
    val isGroupOwner: Boolean,
    val isOpened : Boolean,
){
    fun toModel() = GroupCapsuleMember(
        memberId = this.id,
        nickname = this.nickname,
        profileUrl = this.profileUrl,
        isGroupOwner = this.isGroupOwner,
        isOpened = this.isOpened
    )
}
